lib/api/CDataFrameAnalysisRunner.cc

```diff
@@ -158,19 +153,30 @@ std::size_t CDataFrameAnalysisRunner::estimateMemoryUsage(std::size_t numberRows
            this->estimateBookkeepingMemoryUsage(m_NumberPartitions, numberRows, numberColumns);
 }
 
+void CDataFrameAnalysisRunner::recordProgress(double fractionalProgress) {
+    m_FractionalProgress.fetch_add(std::max(
+        static_cast<int>(MAXIMUM_FRACTIONAL_PROGRESS * fractionalProgress + 0.5), 1));
```

Comment:
Where we've had progress indicators previously we've rounded down rather than to the nearest integer, because we don't want the progress to go to 100% when it's completely finished.